File: extract_lessons.js
const fs = require('fs');
const path = require('path');

const tsFile = path.resolve('d:/statlab/deploy_bundle/client/src/GenAiCourseData.ts');
const jsonFile = path.resolve('d:/statlab/deploy_bundle/server/data/gen_ai_lessons.json');

const content = fs.readFileSync(tsFile, 'utf8');

// The array starts after 'export const GEN_AI_LESSONS: GenAiLesson[] = ['
const startMarker = 'export const GEN_AI_LESSONS: GenAiLesson[] = ';
const startIndex = content.indexOf(startMarker);

if (startIndex !== -1) {
    let bracketCount = 0;
    let endIndex = -1;
    const arrayPart = content.substring(startIndex + startMarker.length);
    
    // Find matching bracket for the array start '['
    let arrayStart = arrayPart.indexOf('[');
    for (let i = arrayStart; i < arrayPart.length; i++) {
        if (arrayPart[i] === '[') bracketCount++;
        else if (arrayPart[i] === ']') bracketCount--;
        
        if (bracketCount === 0) {
            endIndex = i + 1;
            break;
        }
    }
    
    if (endIndex !== -1) {
        const lessonsStr = arrayPart.substring(arrayStart, endIndex);
        // Use eval safely-ish on a data object (caution: only because I know the content)
        // Or better: just use a tool to convert JS object string to JSON
        try {
            // We need to strip some TS types if any, but GEN_AI_LESSONS is mostly data
            // We can wrap it in an object to eval
            const lessons = eval(`(${lessonsStr})`);
            
            if (!fs.existsSync(path.dirname(jsonFile))) {
                fs.mkdirSync(path.dirname(jsonFile), { recursive: true });
            }
            fs.writeFileSync(jsonFile, JSON.stringify(lessons, null, 2), 'utf8');
            console.log('Successfully exported lessons to JSON');
        } catch (err) {
            console.error('Eval failed:', err);
        }
    } else {
        console.error('Could not find end of array');
    }
} else {
    console.error('Could not find GEN_AI_LESSONS');
}

File: extract_lessons.py
import json
import re

ts_file = r'd:\statlab\deploy_bundle\client\src\GenAiCourseData.ts'
json_file = r'd:\statlab\deploy_bundle\server\data\gen_ai_lessons.json'

with open(ts_file, 'r', encoding='utf-8') as f:
    content = f.read()

# Extract GEN_AI_LESSONS array
# This is tricky because it's a complex TypeScript object. 
# I'll use a regex to find the start and end of the array, then I'll use a simple parser or just manual extraction if it's not too long.
# Actually, I'll use a python script that can "evaluate" or at least roughly parse the JS object.

# For now, I'll try to find the start of the array
start_marker = "export const GEN_AI_LESSONS: GenAiLesson[] = ["
start_idx = content.find(start_marker)
if start_idx != -1:
    # Find matching closing bracket for the array
    # We can use a simple counter for brackets
    bracket_count = 0
    end_idx = -1
    for i in range(start_idx + len(start_marker) - 1, len(content)):
        if content[i] == '[':
            bracket_count += 1
        elif content[i] == ']':
            bracket_count -= 1
            if bracket_count == 0:
                end_idx = i + 1
                break
    
    if end_idx != -1:
        lessons_str = content[start_idx + len(start_marker) - 1 : end_idx]
        # Clean up some TS-specific things if any, but since it's just data it might be mostly JSON-like
        # I'll use a very basic replacement to make it valid JSON
        # 1. Replace backticks with " (and escape existing ")
        # 2. Add quotes to keys
        # Actually, it might be easier to just read the file and find the objects.
        
        # Alternative: Let's assume it's well-formatted and just extract it carefully.
        # But wait, I have a better idea. I'll just write a small Node script to import and export it to JSON.
        pass

# I'll use Node to extract the data because it understands TS-like objects better if I just treat it as JS.
